### PR TITLE
Fix partial fasta generation

### PR DESCRIPTION
**Pull request by timodonnell**
This builds on #451 to fix partial fasta generation

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/guacamole/454)

### CODE REVIEW

File: scripts/generate-partial-fasta-for-tests [hunk 1]
```diff
@@ -3,14 +3,18 @@
...
-HG19_FULL_FASTA_PATH=/path/to/ucsc.hg19.fasta
-GRCH37_FULL_FASTA_PATH=/path/to/human_g1k_v37.fasta
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 /path/to/ucsc.hg19.fasta /path/to/human_g1k_v37.fasta"
+    exit 1
+fi
 
-set -e
-set -x
+HG19_FULL_FASTA_PATH=$1
```
Comment (ryan-williams) [inline scripts/generate-partial-fasta-for-tests]: we should quote bash vars where appropriate

File: scripts/generate-partial-fasta-for-tests [hunk 2]
```diff
@@ -27,15 +32,20 @@ else
...
 fi
 
+set -e
+set -x
+
+full_jar=$jar:target/scala-2.10.5/test-classes
+
 # HG19 reference
-time java -Xmx6g -XX:MaxPermSize=512m "-Dspark.master=local[1]" -cp $jar org.hammerlab.guacamole.other_entrypoints.GeneratePartialFasta \
+time java -Xmx6g -XX:MaxPermSize=512m "-Dspark.master=local[1]" -cp $full_jar $MAIN_CLASS \
```
Comment (ryan-williams) [inline scripts/generate-partial-fasta-for-tests]: ditto re: quotes

File: src/test/scala/org/hammerlab/guacamole/main/GeneratePartialFasta.scala [hunk 3]
```diff
@@ -68,13 +80,14 @@ object GeneratePartialFasta extends Logging {
...
     val writer = new BufferedWriter(new FileWriter(fd))
+
     for {
       contig <- loci.contigs
       SimpleRange(start, end) <- contig.ranges
     } {
       try {
         val sequence = Bases.basesToString(reference.getContig(contig.name).slice(start.toInt, end.toInt))
-        writer.write(">%s:%d-%d/%d\n".format(contig, start, end, contigLengths(contig.name)))
+        writer.write(">%s:%d-%d/%d\n".format(contig.name, start, end, contigLengths(contig.name)))
```
Comment (ryan-williams) [inline src/test/scala/org/hammerlab/guacamole/main/GeneratePartialFasta.scala]: good catch, thanks

**Comment by timodonnell** [comment]
Cool, updated to quote bash variables in script

**Comment by ryan-williams** [comment]
lgtm!

**Pull request merged**